packages/react/src/components/AccountSettings/ConfigureTOTP/ConfigureTOTP.tsx

```diff
@@ -0,0 +1,167 @@
+  const [qrCode, setQrCode] = React.useState<string>(null);
+  const [formValues, setFormValues] = React.useState<FormValues>({ code: '' });
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  const { user, isLoading } = useAuth();
+
+  const hasInit = React.useRef(false);
+
+  const generateQRCode = React.useCallback(
+    async (user: AmplifyUser): Promise<void> => {
+      try {
+        const newSecretKey = await setupTOTP(user);
+        const username = totpUsername || user?.username;
+        const totpCode = getTotpCodeURL(totpIssuer, username, newSecretKey);
+        const qrCodeImageSource = await QRCode.toDataURL(totpCode);
+
+        if (!hasInit.current) {
+          setSecretKey(newSecretKey);
+          setQrCode(qrCodeImageSource);
+          hasInit.current = false;
+        }
+      } catch (e) {
+        logger.error(e);
+      }
+    },
+    [totpIssuer, totpUsername]
+  );
+
+  React.useEffect(() => {
+    if (user && !hasInit.current) {
+      generateQRCode(user);
+    }
+  }, [generateQRCode, user]);
+
+  // translations
+  const confirmText = translate('Confirm');
+  const copyCodeText = translate('Copy Secret Code');
+
+  // event handlers
+  const handleChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      event.preventDefault();
+      const { name, value } = event.target;
+      setFormValues((formValues) => ({ ...formValues, [name]: value }));
+    },
+    []
+  );
+
+  const runVerifyTotpToken = React.useCallback(
+    async ({ user, code }: { user: AmplifyUser; code: string }) => {
+      if (errorMessage) {
+        setErrorMessage(null);
+      }
+      try {
+        await verifyTOTPToken({ user, code });
+
+        onSuccess?.(); // notify success to the parent
+      } catch (e) {
+        const error = e as Error;
+        if (error.message) {
+          setErrorMessage(error.message);
+        }
+
+        onError?.(error); // notify error to the parent
+      }
+    },
+    [errorMessage, onError, onSuccess]
+  );
+
+  const handleSubmit = React.useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      const { code } = formValues;
+      runVerifyTotpToken({ user, code });
+    },
+    [user, formValues, runVerifyTotpToken]
+  );
+
+  const handleCopy = React.useCallback(() => {
+    navigator.clipboard.writeText(secretKey);
+  }, [secretKey]);
+
+  /** Return null if user isn't authenticated in the first place */
+  if (!user) {
+    logger.warn('<ConfigureTOTP /> requires user to be authenticated.');
+    return null;
+  }
+
+  /** Return null if Auth.getCurrentAuthenticatedUser is still in progress  */
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <View as="form" onSubmit={handleSubmit}>
+      <Flex direction="column" alignItems="center">
+        {qrCode ? (
+          <SecretKeyQRCode
+            data-amplify-qrcode
```

Comment:
Unless we really need to, let's avoid using `data-` attributes

Reply:
Agreed, it's not like it has its own style anyway

Reply:
```suggestion
```